sound/soc/sof/intel/hda-dai.c

```diff
@@ -33,6 +33,43 @@ struct hda_pipe_params {
 	unsigned int link_bps;
 };
 
+int hda_dai_config(struct snd_sof_dev *sdev,
+					struct snd_soc_dai_link *link,
+					struct sof_ipc_dai_config *config,
+					int direction)
+{
```

Comment:
alignment is out of whack?

Reply:
yes, I will refine it later. I just finish this RFC before leaving office. Sorry, it is not perfect.

Reply:
Also better if the masks and slots can be embedded in a new structure, that way you are only passing a single ptr to a structure rather than 4 ptrs to ints.